### PR TITLE
Add accommodation merge history

### PR DESCRIPTION
**Pull request by oleg-n-k**
Add accommodation merge history

### CODE REVIEW

File: HappyTravel.Nakijin.Api/Services/IAccommodationMergeHistoryService.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+using HappyTravel.Nakijin.Data.Models;
+using HappyTravel.Nakijin.Data.Models.Accommodations;
+
+namespace HappyTravel.Nakijin.Api.Services;
+
+public interface IAccommodationMergeHistoryService
+{
+    void Collect(RichAccommodationDetails sourceAccommodation, RichAccommodationDetails mergedAccommodation);
+    Task<List<AccommodationMergeHistoryEvent>> GetBySourceIdFromDb(int sourceAccommodationId, CancellationToken cancellationToken);
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Services/IAccommodationMergeHistoryService.cs:12]: why this two methods needed?

File: HappyTravel.Nakijin.Api/Services/AccommodationMergeHistoryService.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            Created = utcNow
+        };
+    }
+    
+
+    private (int id, Dictionary<string, string> supplierCodes) GetData(RichAccommodationDetails accommodation)
+        => (accommodation.Id, accommodation.SupplierAccommodationCodes);
+
+
+    private readonly Queue<AccommodationMergeHistoryEvent> _eventStorage = new();
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Services/AccommodationMergeHistoryService.cs:81]: why `Queu` ?

**Review by hasmikharutyunyan — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Services/AccommodationMergeHistoryService.cs (hunk 2 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Services/AccommodationMergeHistoryService.cs:81]: Actually it doesn't matter

File: HappyTravel.Nakijin.Api/Services/IAccommodationMergeHistoryService.cs (hunk 1 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Services/IAccommodationMergeHistoryService.cs:12]: They will be useful for the un-merge process;

File: HappyTravel.Nakijin.Api/Services/AccommodationMergeHistoryService.cs (hunk 2 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Services/AccommodationMergeHistoryService.cs:81]: IEnumerable needed

**Pull request merged**